service/collector_test.go

```diff
@@ -135,6 +136,19 @@ func TestCollector_Start(t *testing.T) {
 	}, time.Second*2, time.Millisecond*200)
 }
 
+func TestCollector_Start(t *testing.T) {
+	testCollectorStartHelper(t)
+}
+
+func TestCollector_StartWithOtelInternalMetrics(t *testing.T) {
+	featuregate.Register(featuregate.Gate{
```

Comment:
note that setting a feature-gate during a unit test affects all other unit tests as well.  If you want to set it during a test only, you need to set it back at the end.

@Aneurysm9 I've added a test util for my feature gate here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7177/files#diff-a4da287cd470d6d0ff4cdcd03a3d5cf4a5ccb76c9dea73d4cc961bde6819999fR34 for setting/resetting it during tests.  I could make that more generic and add it to the featuregate package if you think that would be helpful for others.

Reply:
That could be useful.  To this point I believe that all of the feature gates we've added have been well localized and included in the component configuration as unexported fields with the `featuregate.IsEnabled()` call in `createDefaultConfig()`.  This meant that testing could happen without involving the feature gate registry by simply providing different configuration.  Your use is a bit more invasive, though, so I see where that approach makes more sense.  I'm not sure which style will end up being more common in the end.

Reply:
@dashpole here is what I was pushing to do https://github.com/open-telemetry/opentelemetry-collector/pull/4108#discussion_r726260681 to enable easier testing. You can pass then a "testing registry" to not have to deal with globals. Global state also does not allow parallel testing.

Reply:
I have set and unset the flag for this specific test for now. If a more generic approach is needed to apply for testing with gate lets do it in a separate issue.